webapp/channels/src/components/root/performance_reporter_controller.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useRef} from 'react';
+import {useStore} from 'react-redux';
+
+import {Client4} from 'mattermost-redux/client';
+
+import PerformanceReporter from 'utils/performance_telemetry/reporter';
+
+export default function PerformanceReporterController() {
+    const store = useStore();
+
+    const reporter = useRef<PerformanceReporter>();
+
+    useEffect(() => {
+        reporter.current = new PerformanceReporter(Client4, store);
+        reporter.current.observe();
+
+        // There's no way to clean up web-vitals, so continue to assume that this component won't ever be unmounted
```

Comment:
No, it shouldn't because this component should never be unmounted. If it was unmounted, I think it would create multiple copies of the reporter, but I don't think that should ever happen